lib/will_paginate/view_helpers/link_renderer.rb

```diff
@@ -118,8 +118,10 @@ def symbolized_update(target, other)
         other.each do |key, value|
           key = key.to_sym
           existing = target[key]
-          
-          if value.is_a?(Hash) and (existing.is_a?(Hash) or existing.nil?)
+
+          is_params = lambda { |obj| obj.is_a?(Hash) || (defined?(ActionController::Parameters) && obj.is_a?(ActionController::Parameters)) }
```

Comment:
Rails 5 only supports Ruby >= 2.2.2, so using the old lambda syntax probably isn't necessary.